internal/manifest/referencevalues.go

```diff
@@ -11,11 +11,17 @@ import (
 	"strconv"
 )
 
-// EmbeddedReferenceValuesJSON contains the embedded reference values in JSON format.
-// At startup, they are unmarshaled into a globally-shared ReferenceValues struct.
-//
-//go:embed assets/reference-values.json
-var EmbeddedReferenceValuesJSON []byte
+var (
+	// EmbeddedPlatformRuntimeMappingJSON is a 1:1 mapping of platform names to their runtime component hashes.
+	//
+	//go:embed assets/platform-runtime-mapping.json
+	EmbeddedPlatformRuntimeMappingJSON []byte
+
+	// EmbeddedReferenceValuesJSON contains the embedded reference values in JSON format.
+	//
+	//go:embed assets/reference-values.json
+	EmbeddedReferenceValuesJSON []byte
```

Comment:
I propose we make this a `map[string]ReferenceValues`, mapping the runtime handler name to the reference values of that platform. We can then drop the PlatformRuntimeMapping completely and instead iterate over the keys of this map and look for the known platform strings in the runtime handlers.

In a future PR, we can then transform this into a `map[string][]ReferenceValues`, allowing us to embed multiple values per platform, while transforming the ReferenceValues struct into something platform independent

```go
type ReferenceValues struct {
    SNP
    TDX
}
```

Reply:
I definitely agree to making `ReferenceValues` a platform-independent structure like you described.

Looking at how an implementation of `map[string]ReferenceValues` would look like, I'm not that big of a fan. This would require us to produce the runtime-handler values for each platform in Nix, as well as parsing the platforms out of the runtime handler names in Go. Both of these are doable, but I think the complexity for doing so outweighs the additional binary size costs of embedding the mapping.

Reply:
Decided to just go for the single embedding suggested by @katexochen. I think with a clever implementation, we can build this with reasonable complexity

Reply:
I don't particularly like the idea of extracting the platform the runtime handler name, but I also don't have any better ideas, so it's fine with me ig.